### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by prcutler**
Related to discussion #67, Fixes issue #76

I started by making some small grammar and typo fixes on a couple of pages.  I also added a note not to share or commit your token.

I then renamed the `fetching_data.md` page that uses the REPL to differentiate it from the page I wrote that also shows some of the returns when using a print statement.

Before I write up how to fetch your data in your personal collection, please share any feedback.  I'm undecided if I should split it into two pages or not.  If it doesn't meet your standards, please feel free to let me know and I will re-write and edit it.  You won't hurt my feelings. :)

Thanks again for this great library and especially all the help learning it.

### CODE REVIEW

**Comment by JOJ0** [comment]
Unfortuantely it's not possible to let the documentation render online on readthedocs.io to see what it would finally look like. In any case make yourself comfortable with the index files written in restructured text format: https://github.com/joalla/discogs_client/blob/master/docs/source/index.rst

We use Sphinx to generated the final doc. What you _could_ do to see the rendered final docs is install Sphinx locally and then just open the generated html files locally. You find the requirments for that in sphinx_requirements.txt. Once you installed try to generate the docs using `make html` inside the docs root dir `docs/` Get back to me whether that works I am not sure if I got this right from the top of my head.

**Comment by prcutler** [comment]
It's been a few years since I used rST and Sphinx, but I can send a PR that includes the rST files for Readthedocs.  Sorry for the confusion, the [comment here in #64](https://github.com/joalla/discogs_client/discussions/64#discussioncomment-1555616) made me think it would work in markdown.

I'll incorporate your comments above and revise it and send a PR in rST in the next day or two.

Thanks!

**Comment by JOJ0** [comment]
No, it's fine in markdown. The documentation page your wrote is perfectly fine in markdown and we want it that way! We use a markdown plugin for Sphinx that handles that.
It is the index files only that need to be written in .rst. But it's rather self-explanatory. Adding a news file is as easy as just putting the filename into a new line :-)

**Comment by JOJ0** [comment]
Have a look here again, it's all md files that are stated inside this .rst index file ;-) https://github.com/joalla/discogs_client/blob/master/docs/source/index.rst

**Comment by prcutler** [comment]
That makes more sense.  I'll get working on the changes and the index file.  Thanks for clarifying.

File: docs/source/fetching_data_repl.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Fetching data
+# Fetching data using the REPL
 
 Note that the examples in this chapter are shown using a Python REPL and require
 an existing [Client object](quickstart.md) already.
```
[the remark this answers is not in the document]
Comment (prcutler) [reply docs/source/fetching_data_repl.md:4]: I agree - I know I did code examples vs. the REPL, but I think having both is good for how people prefer to learn a new API.  I'll take a look and I'm updating the PR with the typos - thanks for the extra set of eyes.  

In contribution, I made a link to the Ideas section of Discussions, that should help as well.

[the remark this answers is not in the document]
Comment (prcutler) [reply docs/source/fetching_data_repl.md:4]: Here's my suggestion and PR:
```
Note that the examples in this chapter are shown using a Python REPL and require an existing [Client object](quickstart.md) already. Using the Python REPL can be an easy way to test and review data and information available using the Discogs API.
```

**Comment by prcutler** [comment]
Sorry about all the commits, I'm still newish to VS Code and was trying to use that with PRs.  I still need to write the section on querying a user's specific collection.  It might be a few days until I get that one done.

Thanks again for all the feedback.

**Comment by JOJ0** [comment]
> I realized I committed to the main branch in my repository, so the update is listed as the Merge Pull Request above.

That's fine with me, no worries, but a quick note on git usage because I think you might appreciate learning something new every day :-)

Not sure how you did this "merge request", maybe vscode did that for you. But in any case, you don't need to send merge or pull requests to yourself. You could have just merged your two branches using a few git commands and this PR would have been magically updated with the correct commits:

```
git checkout master
git merge docs-update-dec21
git push
```

Then continue just working in your master branch. Since you opened the PR from your master branch to the joalla master branch, this is perfectly fine.

**Comment by JOJ0** [comment]
I am not sure if you managed to build the docs and see what they finally look like and whether everything is displayed in the inventory and indexes properly. These are the steps necessary I guess (Also something that's probably missing in our docs, but rather in the contribution section I think):
```
# launch your development venv and install sphinx requirements into it:
pip install -r sphinx_requirements.txt
git checkout your_feature_branch  # or just master in your case
python setup.py develop  # the package needs to be installed, otherwise Sphinx won't find it
cd docs
make html  # shouldn't throw too much "red" lines
# Open the built docs in a browser, on Linux that probably would be something like that:
xdg-open build/html/index.html
```
It helps me personally to often have a look at the final generated docs to really see what it looks like and spotting final errors.

**Comment by JOJ0** [comment]
Oh there is quite some "red" during the make command. Please ignore lines like this:
```
docstring of discogs_client.models.ListItem.id:10: WARNING: Unexpected indentation.
docstring of discogs_client.models.ListItem.comment:10: WARNING: Unexpected indentation.
docstring of discogs_client.models.ListItem.display_title:10: WARNING: Unexpected indentation.
docstring of discogs_client.models.ListItem.type:10: WARNING: Unexpected indentation.
docstring of discogs_client.models.ListItem.image_url:10: WARNING: Unexpected indentation.
```

but consider any other red lines / get back to me with questions. Thanks!

**Comment by JOJ0** [comment]
please rename the file fetching_data_code.md to just fetching_data.md

File: docs/source/fetching_data_repl.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JOJ0) [reply docs/source/fetching_data_repl.md]: Using the Python REPL can be an easy way to test and review the information discogs_client is retrieving from the Discogs API.

You might want to rephrase that again, but would I'd like to get across the net ist that this is the way to "see what the data looks like when presented through discogs_client"

**Comment by JOJ0** [comment]
> Sorry about all the commits, I'm still newish to VS Code and was trying to use that with PRs. I still need to write the section on querying a user's specific collection. It might be a few days until I get that one done.
> 
> Thanks again for all the feedback.

No worries, as mentioned somewhere else already :-) I use vscode as well but can't get my head around clicking so much things when I could just use plain old git commands. I think understanding git on the cli first is key to be comfortable and understand what guis are doing. But that might be just my opinion. Work away, you'll understand what it does at some point! :-)

**Comment by prcutler** [comment]
Thanks for the feedback - I'm going back to using the CLI again!  It's been almost 10 years since I've been active in an open source project and I'm a bit rusty.  Appreciate the tips!

I'll get these fixes done and build it locally too.

**Comment by prcutler** [comment]
Just tried to build the docs and it fails.  Any hints?  Sorry to be so much trouble.

```
discogs_client/docs on  master [?] via 🐍 v3.10.0 (.venv) 
❯ make html
Traceback (most recent call last):
  File "/Users/prcutler/workspace/discogs_client/.venv/bin/sphinx-build", line 5, in <module>
    from sphinx.cmd.build import main
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/application.py", line 32, in <module>
    from sphinx.config import Config
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/config.py", line 23, in <module>
    from sphinx.util import logging
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/util/__init__.py", line 35, in <module>
    from sphinx.util import smartypants  # noqa
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/util/smartypants.py", line 33, in <module>
    from sphinx.util.docutils import __version_info__ as docutils_version
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/util/docutils.py", line 31, in <module>
    from sphinx.util.typing import RoleFunction
  File "/Users/prcutler/workspace/discogs_client/.venv/lib/python3.10/site-packages/sphinx/util/typing.py", line 34, in <module>
    from types import Union as types_Union
ImportError: cannot import name 'Union' from 'types' (/usr/local/Cellar/python@3.10/3.10.0_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/types.py)
make: *** [html] Error 1

discogs_client/docs on  master [?] via 🐍 v3.10.0 (.venv) 
❯ 
```

**Comment by JOJ0** [comment]
> Just tried to build the docs and it fails. Any hints? Sorry to be so much trouble.

No trouble at all! We appreciate all your efforts to improve docs and find possible problems like this one for future contributors! :-)

>     from types import Union as types_Union
> ImportError: cannot import name 'Union' from 'types' (/usr/local/Cellar/python@3.10/3.10.0_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/types.py)
> make: *** [html] Error 1
> 
> discogs_client/docs on  master [?] via 🐍 v3.10.0 (.venv) 

Seems you are using Python 3.10. Any chance you could try it with Python 3.9? Will try 3.10. if I find the time though. Maybe Sph8inx doesn't like that (yet).

Also: Did all your pip install commands go through flawlessly? Send a pip list please.

**Comment by prcutler** [comment]
It was Python 3.10 (of course!)

I know I'm saying it alot, but thanks.

**Comment by prcutler** [comment]
I may have been distracted building my project now that I know the API. :) I think I have all changes you asked for - I was able to get it to build. I fixed a couple of the errors it showed me and that I found.  I also reworked that opening paragraph for the REPL chapter again, so take a look and let me know your thoughts.

I still need to write the section on querying a user's collection data, I haven't forgot.  Also, if there is any documentation that needs to be written or edited, feel free to mention me or assign issues to me, I'm happy to help.

**Comment by JOJ0** [comment]
Hi @prcutler, sorry for the rather long silence here. I think I was facing exactly that kind of distraction you are describing :-)

Thank you so much for your your efforts. I just built the docs and are now reviewing the final html. I am sure we will get this release-ready soon. Also thank you very much for the offer for future docs. I am sure there is plenty of opportunity. Most of the time as you probably are aware of already, things that need documenting come out of github discussions or issues. Feel free to suggest in a discussion that there is a need by mentioning me and I'll try to do the same by pointing you there! Thanks again! That's awesome news!

File: docs/source/fetching_data.md [hunk 2]
```diff
@@ -1,161 +1,83 @@
...
-```python
->>> artist.releases.page(1)
-[...]
+['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_known_invalid_keys', 'artists', 'changes', 'client', 'companies', 'country', 'credits', 'data', 'data_quality', 'delete', 'fetch', 'formats', 'genres', 'id', 'images', 'labels', 'marketplace_stats', 'master', 'notes', 'refresh', 'save', 'status', 'styles', 'thumb', 'title', 'tracklist', 'url', 'videos', 'year']
 ```
 
-## Release
-
-Query for a release using its Discogs ID:
+###Artists
```
Comment (JOJ0) [inline docs/source/fetching_data.md:22]: I think there is a space missing, it doesn't render correctly. Also I suggest you always use full code blocks instead of "inline single tick" code markup because the readthedocs template renders these very different (red font) and it's easier to read if everything has the boxes with grey background. Have a look on the final render, that's easier than explaining with words:


Screen Shot 2022-01-14 at 07 54 08

File: docs/source/fetching_data.md [hunk 3]
```diff
@@ -1,161 +1,83 @@
...
-Get a list of all `Artist`s associated with this `Release`:
-
-```python
->>> release.artists
-[<Artist "...">]
+[<Artist 2672269 'Poliça'>, <Artist 4771951 'Stargaze (4)'>] Music For The Long Emergency
 ```
 
-Get the tracklist for this `Release`:
+Data returns JSON including the URL to the page for the given release:
```
Comment (JOJ0) [inline docs/source/fetching_data.md:37]: Technically it's a dict. Requests library already translated it from json string to a python dict.

File: docs/source/fetching_data.md [hunk 4]
```diff
@@ -1,161 +1,83 @@
...
 ```
+print(d.release(20017387).artists[0].name, type(d.release(1443762).artists))
+```
+returns a list of Artists associated with the release:
 
-### Special properties
+`Chvrches <class 'list'>`
 
-Get the title of this `Release`:
+This returns a list as some releases have multiple artists:
```
Comment (JOJ0) [inline docs/source/fetching_data.md:31]: And then rephrase this to:

Artists are always returned as a list as some releases have multiple artists:

File: docs/source/fetching_data.md [hunk 5]
```diff
@@ -1,161 +1,83 @@
...
->>> label.sublabels
-[...]
-```
 
-Get the `Label`'s parent label, if it exists:
-
-```python
->>> label.parent_label
-<Label "Warp Records Limited">
-```
```
Comment (JOJ0) [inline docs/source/fetching_data.md:161]: A chapter about labels would be nice since this was asked often in issues/discussions: How to get the catalog number. It's hidden in the labels dict data. Similar story as .artists and .format:

- .labels is a list
- catalog numbers are hidden inside each single label's data dict (each label that released often has it's own catalog number). Eg: `.labels[0].data['catno']`

(and yes it whould be a nice addition to the library if the catno could be accessed like .labels[0].catno` - I want to make a pull for this for ages already ;-) Not your concern right, just reminding myself :-))

**Review by JOJ0 — CHANGES_REQUESTED** [review]
Alright, some change requests. I do love how the new chapters list now looks like. I think it's structured well. Also the chapter you wrote starting with exploring with dir() and stuff will be very helpful to new users and will certainly help us maintainers as well. We'd just have to point to this well written docs when stuff is asked in discussions. Love it! Thanks!

Screen Shot 2022-01-14 at 08 40 42

**Comment by prcutler** [comment]
Thanks for the feedback.  I'll get the changes done this weekend.  And I still owe more docs, other projects have distracted me.  :)

**Comment by prcutler** [comment]
Thank you!  I'll get the fixes done in the next few days.

**Comment by JOJ0** [comment]
> Thank you! I'll get the fixes done in the next few days. I may open a new issue for the new chapter.

Hi, yes please leave out the Collection chapter for now and let's focus on finishing the pending changes in this PR first.
Also please don't include the chapter about labels I suggested in this PR (if you haven't started yet) because it was my suggestion and is something new again. I might even write this chapter myself or wait until a feature around this is implemented and write the docs afterwards.
Thanks for your efforts!

**Comment by prcutler** [comment]
Ok - I'll get this done tomorrow, I'm tied up today.

Thanks!

File: docs/source/writing_docs.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Writing and Updating Documentation
+
+You can help keep the documentation up to date by submitting pull requests to the ```python3-discogs-client``` library on Github. For more information on this process, please [read Github's documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on forking and creating pull requests.
+
+## Fork and Clone the Repository
+
+The first step is to fork the repository.  In the upper right hand corner of Github, click the ```Fork``` button.  This will create a copy of the repository in your account.
+
+The second step is to clone the ```python3-discogs-client``` library to your computer.  From the ```python3-discogs-client``` fork you just created, press the ```Code``` button and copy the ```git``` URL and clone it in your code editor of choice.  Create a new branch in git and start editing the documentation.
```
Comment (JOJ0) [inline docs/source/writing_docs.md:9]: ....press the Code button, copy the git URL and clone it in your code editor of choice or just use the terminal:

 ```
 git clone <git url>
 ```

File: docs/source/writing_docs.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (JOJ0) [inline docs/source/writing_docs.md:10]:
```
git checkout -b my_docs_additions_branch
```

File: docs/source/contribution.md [hunk 8]
```diff
@@ -40,3 +38,7 @@ source:
 ```
 pip install -r requirements.txt --update
 ```
+
+## Writing Documentation
+
+You can help keep the documentation current by editing or adding new pages to the documentation.  Please see the [Writing Documentation page](writing_docs.md) for more information on how to clone the repository, edit the documentation, and build the documentation.
```
Comment (JOJ0) [inline docs/source/contribution.md:44]: I'm still thinking about whether or not the docs chapter should be filed under the "contribution" chapter, but on the other hand, I like having such a nice separate major chapter for docs only. Also I like that you explain some git basics there since I think we can assume that people that contribute actual code to a library probably know the git basics already but newish users to python hacking and using a library might not. Giving them a nice starting point for improving unclear things in the docs is a good thing. Well done Peter! Let's get this merged! Just a few changes requested.

**Review by JOJ0 — CHANGES_REQUESTED** [review]

**Comment by prcutler** [comment]
I've got a 3.9 venv setup, but I"m getting an error trying to build the docs:

```
make html
Running Sphinx v4.4.0
WARNING: sphinx_rtd_theme (< 0.3.0) found. It will not be available since Sphinx-6.0
```

The changes are in the updated PR, I just haven't verified them.

File: docs/source/writing_docs.md [hunk 9]
```diff
@@ -6,11 +6,21 @@ You can help keep the documentation up to date by submitting pull requests to th
...
+```
+git clone <git url>
+```
+
+  Create a new branch in git and start editing the documentation:
+
+  ```
+  git checkout -b my_docs_changes_branch
+  ```
```
Comment (JOJ0) [inline docs/source/writing_docs.md:19]: and here too.

**Review by JOJ0 — APPROVED** [review]

**Comment by JOJ0** [comment]
Thanks again @prcutler for these great additions and refinements to our docs!
